pytorch_lightning/trainer/trainer.py

```diff
@@ -447,21 +447,55 @@ def default_attributes(cls):
 
         return args
 
+    @classmethod
+    def get_init_arguments_and_types(cls) -> List[Tuple[str, Set, Any]]:
+        r"""Scans the Trainer signature and returns argument names, types and default values.
+
+        Returns:
+            List with tuples of 3 values:
+            (argument name, set with argument types, argument default value).
```

Comment:
can we add doctest with some limited print out
```
>>> args = Trainer.get_init_arguments_and_types()
>>> import pprint
>>> pprint.pprint(args)  # doctest: +ELLIPSIS
now here print cople argument and the remenaing ignore by "..."
```
ping me if you need some help here

Reply:
if you decide to add it make it as wip until you finish it... :]

Reply:
@Borda there are no any guaranties, that this test won't break every time, when signature of the Trainer changed. Even small spelling could broke this doctest. Isn't too strict?

Reply:
Maybe something like this?
```
        >>> import inspect
        >>> args = Trainer.get_init_arguments_and_types()
        >>> arg_names = [arg[0] for arg in args]
        >>> sign_names = [arg for arg in inspect.signature(Trainer).parameters]
        >>> assert set(arg_names) == set(sign_names)
```

Reply:
you do not need assert, it is compared to the output and ls this serves as example in docs so assert does not say much...

Reply:
did the edit